src/main/java/ng/abdlquadri/eventbus/EventBusFrameHandler.java

```diff
@@ -46,7 +47,8 @@ public void run() {
 
   @Override
   public void channelInactive(ChannelHandlerContext ctx) throws Exception {
-    log.log(Level.SEVERE, "CHANNEL NOT ACTIVE");
+	// log OR throw exception, never both
```

Comment:
Well a key requirement for this library is Android 4.0+., Which can not run Java8 that vertx depends on.

You can create an issue regarding all suggestions you have. Myself/Someone will find it interesting and work on it. Tjhanks!